### PR TITLE
[Magiclysm] Make workshop elves spawn in a workshop

### PR DESCRIPTION
**Pull request by Theawesomeboophis**
#### Summary
Mods "Adds new location for workshop elves."

#### Purpose of change
Make workshop elves spawn in new location called elf workshop.

#### Describe the solution
Adds elf workshop and makes it so workshop elves spawn there.

#### Describe alternatives you've considered
Making workshop elves non-hostile to forge monsters.

#### Testing
All errors fixed, looks and works as intended.

### CODE REVIEW

**Comment by Theawesomeboophis** [comment]
Testing to see if lint fixed the end of string bug.

**Comment by Theawesomeboophis** [comment]
DEBUG    : Error: Json error: data/mods/Magiclysm/monstergroups.json:199:5: requested non-existing member "monster" in {
        "type": "monstergroup",
        "name": "GROUP_WORKSHOP",
        "default": "mon_workshop_elf",
        "monsters": [ { "monster": "mon_workshop_elf", "freq": 100, "cost_multiplier": 2 } ]
      }
    

        "monsters": [ { "monster": "mon_workshop_elf", "freq": 100, "cost_multiplier": 2 } ]
      }
    ]
    ^

  }
]


 FUNCTION : bool main_menu::new_character_tab()
 FILE     : src/main_menu.cpp
 LINE     : 952
new bug has appeared.

**Comment by Theawesomeboophis** [comment]
found issue, monstergroup is plural.

**Comment by Theawesomeboophis** [comment]
Yet another error to fix
 DEBUG    : Error: Json error: data/mods/Magiclysm/monstergroups.json:197:20: expected string but got '['

        "name": "GROUP_WORKSHOP",
        "default": "mon_workshop_elf",
        "monster": [
                   ^
                     { "monster": "mon_workshop_elf", "freq": 100, "cost_multiplier": 2 } ]
      }
    ]


 FUNCTION : bool main_menu::new_character_tab()
 FILE     : src/main_menu.cpp
 LINE     : 952

**Comment by Theawesomeboophis** [comment]
Not sure how to fix glitch.

File: data/mods/Magiclysm/monstergroups.json [hunk 1]
```diff
@@ -189,7 +189,13 @@
     "monsters": [
       { "monster": "mon_orc_warrior", "freq": 100, "cost_multiplier": 5 },
       { "monster": "mon_orc_archer", "freq": 25, "cost_multiplier": 7 },
-      { "monster": "mon_orc_blood_warrior", "freq": 15, "cost_multiplier": 5 }
+      { "monster": "mon_orc_blood_warrior", "freq": 15, "cost_multiplier": 5 },
+      {
+        "type": "monstergroup",
+        "name": "GROUP_WORKSHOP",
+        "default": "mon_workshop_elf",
+        "monster": [ { "monster": "mon_workshop_elf", "freq": 100, "cost_multiplier": 2 } ]
```
Comment (anothersimulacrum) [inline data/mods/Magiclysm/monstergroups.json:197]: I think you want this
```suggestion
        "monsters": [ { "monster": "mon_workshop_elf", "freq": 100, "cost_multiplier": 2 } ]
```

Comment (anothersimulacrum) [inline data/mods/Magiclysm/monstergroups.json:197]: I think you want this
```suggestion
        "type": "monstergroup",
        "name": "GROUP_WORKSHOP",
        "default": "mon_workshop_elf",
        "monsters": [ { "monster": "mon_workshop_elf", "freq": 100, "cost_multiplier": 2 } ]
```
Or this
```suggestion
        "monster": "mon_workshop_elf", "freq": 100, "cost_multiplier": 2
```

Comment (Theawesomeboophis) [reply data/mods/Magiclysm/monstergroups.json]: will try both to see if they work.

Comment (anothersimulacrum) [reply data/mods/Magiclysm/monstergroups.json]: Both should work (though the first one is probably at least slightly wrong) - it just depends on if you want a new monstergroup here, or just an individual monster added to this monstergroup.

Comment (Theawesomeboophis) [reply data/mods/Magiclysm/monstergroups.json]: I think that they should be their own monstergroup as the previous group is orcs.

Comment (anothersimulacrum) [reply data/mods/Magiclysm/monstergroups.json]: Oh, I totally misunderstood what you're trying to do then. None of this is valid, you somehow got this new monstergroup def stuck into te orc one.

File: data/mods/Magiclysm/monstergroups.json [hunk 2]
```diff
@@ -189,7 +189,12 @@
...
-      { "monster": "mon_orc_blood_warrior", "freq": 15, "cost_multiplier": 5 }
+      { "monster": "mon_orc_blood_warrior", "freq": 15, "cost_multiplier": 5 },
+      {"type": "monstergroup",
+        "name": "GROUP_WORKSHOP",
+        "default": "mon_workshop_elf",
+        "monsters": [ { "monster": "mon_workshop_elf", "freq": 100, "cost_multiplier": 2 } ]
+      }
     ]
   }
 ]
```
Comment (anothersimulacrum) [inline data/mods/Magiclysm/monstergroups.json]: You want this.
```suggestion
      { "monster": "mon_orc_blood_warrior", "freq": 15, "cost_multiplier": 5 }
    ]
  },
  {
    "type": "monstergroup",
    "name": "GROUP_WORKSHOP",
    "default": "mon_workshop_elf",
    "monsters": [ { "monster": "mon_workshop_elf", "freq": 100, "cost_multiplier": 2 } ]
  }
]
```

File: data/mods/Magiclysm/monstergroups.json (hunk 1 above)
Comment (Theawesomeboophis) [reply data/mods/Magiclysm/monstergroups.json]: Does that mean that orcs and workshop monstergroups are not separated properly? Is that the problem.

**Comment by Theawesomeboophis** [comment]
DEBUG    : invalid terrain id "f_table"

 FUNCTION : int_id<T> generic_factory<T>::convert(const string_id<T>&, const int_id<T>&, bool) const [with T = ter_t]
 FILE     : src/generic_factory.h
 LINE     : 467
it also says t_metal and f_anvil are incorrect, what are correct ids?

**Comment by actual-nh** [comment]
> DEBUG : invalid terrain id "f_table"
> 
> FUNCTION : int_id generic_factory::convert(const string_id&, const int_id&, bool) const [with T = ter_t]
> FILE : src/generic_factory.h
> LINE : 467
> it also says t_metal and f_anvil are incorrect, what are correct ids?

I will try to take a look after my medications kick in...

**Comment by anothersimulacrum** [comment]
`f_table` is a furniture, not a terrain. You're probably putting it in the terrain part of the palette, not the furniture part.

**Comment by actual-nh** [comment]
What is `t_metal` supposed to be?

**Comment by Theawesomeboophis** [comment]
Steel

**Comment by Theawesomeboophis** [comment]
Gonna retest now and make sure it actually appears in game.

**Comment by Theawesomeboophis** [comment]
Appears but right wall is missing.

**Comment by actual-nh** [comment]
Aha! I just measured the width of the orc_village - 24 characters. Current number is 29, so 5 too many. Taking out one column was the right idea - it just wasn't sufficient.

**Comment by Theawesomeboophis** [comment]
> Aha! I just measured the width of the orc_village - 24 characters. Current number is 29, so 5 too many. Taking out one column was the right idea - it just wasn't sufficient.

So i have to take out 5 rows and it should be fixed?

**Comment by actual-nh** [comment]
> > Aha! I just measured the width of the orc_village - 24 characters. Current number is 29, so 5 too many. Taking out one column was the right idea - it just wasn't sufficient.
> 
> So i have to take out 5 rows and it should be fixed?

Columns (5 characters from each row) - yes, I'm pretty sure.

**Comment by Theawesomeboophis** [comment]
> > > Aha! I just measured the width of the orc_village - 24 characters. Current number is 29, so 5 too many. Taking out one column was the right idea - it just wasn't sufficient.
> > 
> > 
> > So i have to take out 5 rows and it should be fixed?
> 
> Columns (5 characters from each row) - yes, I'm pretty sure.

Did as suggested, retesting.

**Comment by Theawesomeboophis** [comment]
Right wall has returned but the south wall is still missing.

**Comment by Theawesomeboophis** [comment]
I think it should be fixed now, it seems there were 2 extra rows.

**Comment by Theawesomeboophis** [comment]
Orc village is 24x24, elf workshop was 24x26.

**Comment by actual-nh** [comment]
> South wall still missing.

There are currently 25 rows.

**Comment by Theawesomeboophis** [comment]
> > South wall still missing.
> 
> There are currently 25 rows.

That means I have to remove one more row.

**Comment by Theawesomeboophis** [comment]
Extra row removed, gonna retest and see if the literal invisible wall has returned.

**Comment by Theawesomeboophis** [comment]
Works properly now. and all walls spawn.

**Comment by KorGgenT** [comment]
the workshop is currently an empty square... please put some stuff in there. maybe tables, some craftsman's tools, materials, that kind of thing.

**Comment by actual-nh** [comment]
Some furniture/tools to take a look at; will expand:
* Furniture:
  * f_stool (near `f_workbench`es)
  * f_chair (around a f_table)
  * Tool furniture (furniture-tools.json) - these assume power is available:
    * Wood: f_tablesaw, f_router, f_planer, f_jointer, possibly others
    * Metal/Wood: f_drill_press, f_heavy_lathe, possibly others

**Comment by Theawesomeboophis** [comment]
> the workshop is currently an empty square... please put some stuff in there. maybe tables, some craftsman's tools, materials, that kind of thing.
I will put more furniture as suggested. Which furniture should I add to the workshop?

**Comment by Theawesomeboophis** [comment]
Added more furniture as suggested, should make workshop feel more like a workshop and less like a bland square.

**Comment by actual-nh** [comment]
> Added more furniture as suggested, should make workshop feel more like a workshop and less like a bland square.

The bottom was still blank - I added some workbenches and stools; the workbenches are suitable places to put some item tools and (along with the bigger tools in the top section) materials, as KorGgenT suggested.

**Comment by Theawesomeboophis** [comment]
> > Added more furniture as suggested, should make workshop feel more like a workshop and less like a bland square.
> 
> The bottom was still blank - I added some workbenches and stools; the workbenches are suitable places to put some item tools and (along with the bigger tools in the top section) materials, as KorGgenT suggested. EDIT: Oops; linting JSON.

Which materials should be added?

File: data/mods/Magiclysm/worldgen/elf_workshop.json [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    "type": "mapgen",
+    "method": "json",
+    "om_terrain": "elf_workshop",
+    "object": {
+      "fill_ter": "t_metal_floor",
+      "rows": [
+        "111111111111111111111111",
+        "1a...................ja1",
+        "1tc.....at....d....r.ct3",
+        "1ab..................ct1",
```
Comment (actual-nh) [inline data/mods/Magiclysm/worldgen/elf_workshop.json:28]:
```suggestion
        "1bsw.................ct1",
```
I can't see having an anvil next to a table saw.

File: data/mods/Magiclysm/worldgen/elf_workshop.json [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        "1b...........c........d3",
+        "1dtc........ctc.......a1",
```
Comment (actual-nh) [inline data/mods/Magiclysm/worldgen/elf_workshop.json:30]:
```suggestion
        "1dsw........ctc.......a1",
```

File: data/mods/Magiclysm/worldgen/elf_workshop.json [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        "1a...........c.........1",
+        "1atc...................1",
```
Comment (actual-nh) [inline data/mods/Magiclysm/worldgen/elf_workshop.json:32]:
```suggestion
        "1aws...................1",
```

File: data/mods/Magiclysm/worldgen/elf_workshop.json [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        "1d....................d1",
+        "1......................1",
+        "1btc..........d.......d1",
```
Comment (actual-nh) [inline data/mods/Magiclysm/worldgen/elf_workshop.json:35]:
```suggestion
        "1bws..........d.......d1",
```

File: data/mods/Magiclysm/worldgen/elf_workshop.json [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "om_terrain": "elf_workshop",
+    "object": {
+      "fill_ter": "t_metal_floor",
+      "rows": [
+        "111111111111111111111111",
+        "1a...................ja1",
```
Comment (actual-nh) [inline data/mods/Magiclysm/worldgen/elf_workshop.json:26]:
```suggestion
        "1a..................jsw1",
```

File: data/mods/Magiclysm/worldgen/elf_workshop.json [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+        "1tc.....at....d....r.ct3",
```
Comment (actual-nh) [inline data/mods/Magiclysm/worldgen/elf_workshop.json:27]:
```suggestion
        "1tc.....aw....d....r.ct3",
```

File: data/mods/Magiclysm/worldgen/elf_workshop.json [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+        "1dtc........ctc.......a1",
+        "1a...........c.........1",
+        "1atc...................1",
+        "1d....................d1",
+        "1......................1",
+        "1btc..........d.......d1",
+        "1......................1",
+        "1......................1",
+        "1......................1",
+        "1...ws............sw...3",
```
Comment (actual-nh) [inline data/mods/Magiclysm/worldgen/elf_workshop.json:39]:
```suggestion
        "1...ws............sw...1",
        "1......................1",
        "1......................3",
```
Oops! I didn't notice the door.

**Review by actual-nh — COMMENTED** [review]
A few rearrangements for logic (my grandfather on my father's side did woodworking as a hobby). We may want to have 2 different workbenches - for working with metal vs for working with wood - so that appropriate things can be spawned on them. EDIT: Done re latter ('m' is a metal-task workbench).

**Comment by actual-nh** [comment]
> > The bottom was still blank - I added some workbenches and stools; the workbenches are suitable places to put some item tools and (along with the bigger tools in the top section) materials, as KorGgenT suggested. EDIT: Oops; linting JSON.
> 
> Which materials should be added? Also how do I place item tools onto the workbenches?

Hrm. Definitely `enchanted_wood` for some of the wood-working benches and maybe furniture tools. For the metal-working - I'm thinking a few mithral ingots (`mithril_ingot`) and the itemgroup of `tools_anvil` (admittedly, it's unclear how they're going to be using an anvil without a source of heat!), and maybe some copper/bronze/brass or something like that.

For adding item tools (and materials), something like the below example for anvils - which needs JSON linting!
```JSON
"items": {
  "a": [
           { "item": "tools_anvil", "chance": 50, "repeat": [ 1, 2 ] },
           { "item": "mithril_ingot", "chance": 10 },
           { "item": "scrap_copper", "chance": 40 }
   ]
```
Take a look at the other files in the worldgen directory for other examples. Tools can be found in `data/json/items/tool`.

**Comment by Theawesomeboophis** [comment]
> > > The bottom was still blank - I added some workbenches and stools; the workbenches are suitable places to put some item tools and (along with the bigger tools in the top section) materials, as KorGgenT suggested. EDIT: Oops; linting JSON.
> > 
> > 
> > Which materials should be added? Also how do I place item tools onto the workbenches?
> 
> Hrm. Definitely `enchanted_wood` for some of the wood-working benches and maybe furniture tools. For the metal-working - I'm thinking a few mithral ingots (`mithril_ingot`) and the itemgroup of `tools_anvil` (admittedly, it's unclear how they're going to be using an anvil without a source of heat!), and maybe some copper/bronze/brass or something like that.
> 
> For adding item tools (and materials), something like the below example for anvils - which needs JSON linting!
> 
> ```json
> "items": {
>   "a": [
>            { "item": "tools_anvil", "chance": 50, "repeat": [ 1, 2 ] },
>            { "item": "mithril_ingot", "chance": 10 },
>            { "item": "scrap_copper", "chance": 40 }
>    ]
> ```
> 
> Take a look at the other files in the worldgen directory for other examples. Tools can be found in `data/json/items/tool`.

Added the suggested items to anvil, going to add fireplace or other alternative as source of heat.

**Comment by actual-nh** [comment]
> Added the suggested items to anvil, going to add fireplace or other alternative as source of heat. Is there a smelter in the game or something else like that so that players can use the metal scraps and ingots to craft?

I think it would probably require an `f_magical_forge` which should have `alumentum` as "ammo".

**Comment by Theawesomeboophis** [comment]
> > Added the suggested items to anvil, going to add fireplace or other alternative as source of heat. Is there a smelter in the game or something else like that so that players can use the metal scraps and ingots to craft?
> 
> I think it would probably require an `f_magical_forge` which should have `alumentum` as "ammo".

How do I set it to have ammo, do I just put alumentum in the magical forge like with item tools, or do I have to add another section to the items for ammo?

**Comment by actual-nh** [comment]
> > > Added the suggested items to anvil, going to add fireplace or other alternative as source of heat. Is there a smelter in the game or something else like that so that players can use the metal scraps and ingots to craft?
> > 
> > I think it would probably require an `f_magical_forge` which should have `alumentum` as "ammo".
> 
> How do I set it to have ammo, do I just put alumentum in the magical forge like with item tools, or do I have to add another section to the items for ammo?

Good question... I'm thinking try just putting in first.

**Comment by Theawesomeboophis** [comment]
I have attempted the first approach as discussed above.

**Comment by Theawesomeboophis** [comment]
Elves have an aggression of 12, I wonder why they are so hostile in game?

**Comment by Theawesomeboophis** [comment]
Does PATH_AVOID_DANGER_2 increase hostility?

**Comment by actual-nh** [comment]
> Elves have an aggression of 12, I wonder why they are so hostile in game?

Anything 10+ will do it. Try giving them 0-9 and an appropriate anger trigger like FRIEND_ATTACKED. (See 0.F-dev for more-clear information on the levels - <https://github.com/CleverRaven/Cataclysm-DDA/blob/0.F-dev/doc/MONSTERS.md> - and triggers - <https://github.com/CleverRaven/Cataclysm-DDA/blob/0.F-dev/doc/JSON_FLAGS.md#anger-fear-and-placation-triggers>, thanks to Venera3.)

**Comment by Theawesomeboophis** [comment]
> > Elves have an aggression of 12, I wonder why they are so hostile in game?
> 
> Anything 10+ will do it. Try giving them 0-9 and an appropriate anger trigger like FRIEND_ATTACKED. (See 0.F-dev for more-clear information on the levels - https://github.com/CleverRaven/Cataclysm-DDA/blob/0.F-dev/doc/MONSTERS.md - and triggers - https://github.com/CleverRaven/Cataclysm-DDA/blob/0.F-dev/doc/JSON_FLAGS.md#anger-fear-and-placation-triggers, thanks to Venera3.) Sorry I didn't spot that earlier!

I have added anger triggers and lowered aggression.

**Comment by Theawesomeboophis** [comment]
@KorGgenT This should be ready now that merge conflicts and name is fixed, gonna wait for test to see if something messes up.

**Comment by actual-nh** [comment]
The equipment in #49582 may be of interest.

**Comment by Theawesomeboophis** [comment]
That PR looks great, but I don't want the two buildings feeling too similar.

File: data/mods/Magiclysm/worldgen/elf_workshop.json [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+        { "monster": "GROUP_WORKSHOP", "x": [ 3, 21 ], "y": [ 4, 23 ], "density": 0.2, "repeat": [ 1, 5 ] },
+        { "monster": "GROUP_WORKSHOP", "x": [ 3, 21 ], "y": [ 4, 23 ], "density": 0.2, "repeat": [ 1, 5 ] },
+        { "monster": "GROUP_WORKSHOP", "x": [ 3, 21 ], "y": [ 4, 23 ], "density": 0.2, "repeat": [ 1, 5 ] }
+      ]
+    }
+  },
+  {
+    "type": "overmap_special",
+    "id": "elf_workshop",
+    "overmaps": [ { "point": [ 4, 7, 3 ], "overmap": "elf_workshop_north" } ],
```
Comment (curstwist) [inline data/mods/Magiclysm/worldgen/elf_workshop.json:87]: won't this spawn at z level 3? It should probably be [ 0, 0, 0 ] here.
Also, you need to add a roof level.

File: data/mods/Magiclysm/worldgen/elf_workshop.json [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+    "locations": [ "wilderness" ],
+    "city_distance": [ 20, -5 ],
+    "city_sizes": [ 0, 13 ],
+    "occurrences": [ 0, 9 ]
```
Comment (curstwist) [inline data/mods/Magiclysm/worldgen/elf_workshop.json:91]: [ 0, 1 ], or [ 0, 2 ] would be better.  This is occurrences per overmap chunk.

File: data/mods/Magiclysm/worldgen/elf_workshop.json [hunk 12]
```diff
@@ -0,0 +1,102 @@
+[
+  {
+    "type": "palette",
+    "id": "elf_palette",
+    "terrain": { "3": "t_rdoor_c", "1": "t_strconc_wall", ".": "t_metal_floor_olight" },
```
Comment (curstwist) [inline data/mods/Magiclysm/worldgen/elf_workshop.json:5]: `t_metal_floor_olight` every floor tile will have a light in it. We generally reserve overhead lights for labs, Placing some Magiclsym everlasting torches will be better. `t_metal_floor` is the normal metal floor.

File: data/mods/Magiclysm/worldgen/elf_workshop.json [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+      "r": "f_router",
+      "w": "f_workbench",
+      "m": "f_workbench",
+      "s": "f_stool",
+      "f": "f_demon_forge"
+    },
+    "items": {
+      "a": [
+        { "item": "tools_anvil", "chance": 75, "repeat": [ 1, 2 ] },
+        { "item": "mithril_ingot", "chance": 10 },
```
Comment (curstwist) [inline data/mods/Magiclysm/worldgen/elf_workshop.json:22]: `"items"` can be used for item groups but you'll need to use `"place_loot"` to spawn individual items.

File: data/mods/Magiclysm/worldgen/elf_workshop.json [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+        "1......................1",
+        "1...ws............sw...1",
+        "1......................1",
+        "1...........m..........1",
+        "1f.........sa.........f1",
+        "1a.sm......f.......ms.a1",
+        "111111111111111111111111"
+      ],
+      "palettes": [ "elf_palette" ],
+      "place_monsters": [
```
Comment (curstwist) [inline data/mods/Magiclysm/worldgen/elf_workshop.json:76]: That's a lot of elves.

**Review by curstwist — COMMENTED** [review]
Please include a screen shot of your building including the roof once it's added.  You may want to give it a more interesting shape, right now it takes up the entire square OMT.  You could decrease the building's size and add some landscaping.  Since it's an elves workshop you could also add long workbenches and tables with stools and even some toys or other items that these elves make (maybe some magic items like the copper jewelry).

Shelves or racks will also help fill out the space.

I'm not sure how you checked this in game for natural spawns with the current settings.  I was able to spawn it via debug mode for single OMT's though.

**Comment by Theawesomeboophis** [comment]
I don't think you can see roofs in current implementation, and I don't know
how to submit screenshots, I'll fix some of the things and add the stuff
you suggested, thanks for the help. I'll work on this more this weekend (
as it's usually the only time I'm available).

**Comment by KorGgenT** [comment]
upload a screenshot as if it were a file.

**Comment by Theawesomeboophis** [comment]
How do I take one?

**Comment by Theawesomeboophis** [comment]
I don't know why the checks are failing.

**Comment by KorGgenT** [comment]
I'm going to go with curstwist's recommendation of approval for this one.

File: data/mods/Magiclysm/worldgen/elf_workshop.json (hunk 13 above)
Comment (curstwist) [reply data/mods/Magiclysm/worldgen/elf_workshop.json:22]: This isn't how place_loot works.  It needs to be in the mapgen object entry and it must use x,y coordinates.  item_groups get identified as `group` and single items as `item`.  For example:
```
"place_loot": [ { "item": "scrap_copper", "x": 14, "y": 12, "chance": 100 }, { "group": "tools_anvil", "x": 5, "y": 4, "chance": 100 } ],
```

File: data/mods/Magiclysm/worldgen/elf_workshop.json (hunk 14 above)
Comment (curstwist) [reply data/mods/Magiclysm/worldgen/elf_workshop.json:76]: This still needs a roof mapgen entry as well.  Please see the [JSON_ROOF_MAPGEN.md](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/JSON_Mapping_Guides/JSON_ROOF_MAPGEN.md)

**Review by curstwist — CHANGES_REQUESTED** [review]

**Comment by Theawesomeboophis** [comment]
Thanks, I will work on this more and get everything ( hopefully) working
properly.

**Comment by Theawesomeboophis** [comment]
I'm gonna reconvert to draft until everything is fixed and done properly.

**Comment by Theawesomeboophis** [comment]
I have reconverted to draft.

**Comment by Theawesomeboophis** [comment]
@curstwist I have a question about place_loot, how do I get it to place different loot in specified areas ( ex: at "m")? Thanks in advance.

**Comment by Maleclypse** [comment]
@Theawesomeboophis I noticed there is a question here that got lost in the shuffle.  Can you ask whatever questions you need answers to in #mod-dev in the discord?   I'd like to help you get this into mergeable state.

**Comment by Theawesomeboophis** [comment]
Yeah sure, can I have a link to it?

**Comment by Maleclypse** [comment]
https://discord.gg/jFEc7Yp  sure thing. I think that one should work :)

**Comment by Theawesomeboophis** [comment]
I'll continue work on this in my free time and resubmit when finished.

**Comment by Theawesomeboophis** [comment]
I've gotten this branch up to date and will add roofs and other things when I figure out how.

**Pull request closed without merging**